core/src/main/java/org/mapfish/print/map/image/ImageLayer.java

```diff
@@ -0,0 +1,244 @@
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+
+import javax.annotation.Nonnull;
+import javax.imageio.ImageIO;
+
+import static java.awt.image.BufferedImage.TYPE_INT_ARGB_PRE;
+
+/**
+ * <p>Reads a image file from an URL.</p>
+ *
+ * @author MaxComse on 11/08/16.
+ */
+public final class ImageLayer extends AbstractSingleImageLayer {
+
+    private final ImageParam params;
+    private StyleSupplier<GridCoverage2D> styleSupplier;
+    private ExecutorService executorService;
+
+    /**
+     * Constructor.
+     *
+     * @param executorService the thread pool for doing the rendering.
+     * @param styleSupplier   the style to use when drawing the constructed grid coverage on the map.
+     * @param params          the params from the request data.
+     * @param registry        the metrics registry.
+     */
+    protected ImageLayer(final ExecutorService executorService,
+                       final StyleSupplier<GridCoverage2D> styleSupplier,
+                       final ImageParam params,
+                       final MetricRegistry registry) {
+        super(executorService, styleSupplier, params);   
+        this.params = params;
+        this.styleSupplier = styleSupplier;
+        this.executorService = executorService;
+    }
+
+    /**
+      * <p>Renders a image as layer.</p>
+      * <p>Type: <code>image</code></p>
+      */
+    public static final class ImageLayerPlugin extends AbstractGridCoverageLayerPlugin implements MapLayerFactoryPlugin<ImageParam> {
+        private static final String TYPE = "image";
+        @Autowired
+        private ForkJoinPool forkJoinPool;
+        @Autowired
+        private StyleParser styleParser;
+        @Autowired
+        private MetricRegistry metricRegistry;
+
+        @Override
+        public Set<String> getTypeNames() {
+            return Collections.singleton(TYPE);
+        }
+
+        @Override
+        public ImageParam createParameter() {
+            return new ImageParam();
+        }
+
+        @Nonnull
+        @Override
+        public ImageLayer parse(@Nonnull final Template template,
+                              @Nonnull final ImageParam layerData) throws Throwable {
+
+            String styleRef = layerData.style;
+            return new ImageLayer(this.forkJoinPool,
+                    super.<GridCoverage2D>createStyleSupplier(template, styleRef),
+                    layerData, this.metricRegistry);
+        }
+    }
+
+    /**
+     * The parameters for reading an image file, either from the server or from a URL.
+     */
+    public static final class ImageParam extends AbstractLayerParams {
+         
+         /**
+         * The base URL for the image file.  Used for making request.
+         */
+         public String baseURL;
+         
+         /**
+          * The extent of the image.  Used for placing image on map.
+          */
+         public double[] extent;
```

Comment:
It would be good to have a basic validation of this parameter. For example similar to this one: https://github.com/mapfish/mapfish-print/blob/master/core/src/main/java/org/mapfish/print/map/tiled/osm/OsmLayerParam.java#L133-L136